big_tests/tests/ejabberd_node_utils.erl

```diff
@@ -148,37 +148,71 @@ file_exists(Node, Filename) ->
       ConfigVariable :: atom(),
       Value :: string().
 modify_config_file(CfgVarsToChange, Config) ->
-    Node = distributed_helper:mim(),
-    modify_config_file(Node, "vars-toml.config", CfgVarsToChange, Config).
+    modify_config_file(mim, CfgVarsToChange, Config, toml).
 
--spec modify_config_file(distributed_helper:rpc_spec(), string(), [{ConfigVariable, Value}], ct_config()) -> ok when
+-spec modify_config_file(Host, [{ConfigVariable, Value}], ct_config(), toml | cfg) -> ok when
+      Host :: atom(),
       ConfigVariable :: atom(),
       Value :: string().
-modify_config_file(#{node := Node} = RPCSpec, VarsFile, CfgVarsToChange, Config) ->
-    CurrentCfgPath = current_config_path(Node, Config),
-    {ok, CfgTemplate} = file:read_file(config_template_path(Config)),
-    CfgVarsPath = config_vars_path("vars-toml.config", Config),
-    {ok, DefaultVars} = file:consult(CfgVarsPath),
-    {ok, NodeVars} = file:consult(config_vars_path(VarsFile, Config)),
-    PresetVars = case proplists:get_value(preset, Config) of
-                     undefined ->
-                         [];
-                     Name ->
-                         Presets = ct:get_config(ejabberd_presets),
-                         proplists:get_value(list_to_existing_atom(Name), Presets)
-                 end,
-    CfgVars1 = dict:to_list(dict:merge(fun(_, V, _) -> V end,
-                                       dict:from_list(NodeVars),
-                                       dict:from_list(DefaultVars))),
-    CfgVars = dict:to_list(dict:merge(fun(_, V, _) -> V end,
-                                      dict:from_list(PresetVars),
-                                      dict:from_list(CfgVars1))),
-    UpdatedCfgVars = update_config_variables(CfgVarsToChange, CfgVars),
+modify_config_file(Host, VarsToChange, Config, Format) ->
+    VarsFile = vars_file(Format),
+    NodeVarsFile = ct:get_config({hosts, Host, vars}, Config) ++ "." ++ vars_file(Format),
+    TemplatePath = config_template_path(Config, Format),
+    DefaultVarsPath = config_vars_path(VarsFile, Config),
+    NodeVarsPath = config_vars_path(NodeVarsFile, Config),
+
+    {ok, Template} = file:read_file(TemplatePath),
+    {ok, DefaultVars} = file:consult(DefaultVarsPath),
+    {ok, NodeVars} = file:consult(NodeVarsPath),
+    ct:pal("Node vars: ~p", [NodeVars]),
+    PresetVars = preset_vars(Config, Format),
+
+    TemplatedConfig = template_config(Template, [DefaultVars, NodeVars, PresetVars, VarsToChange]),
+    ct:log("Updated config file: ~s", [TemplatedConfig]),
+
+    RPCSpec = distributed_helper:Host(),
```

Comment:
This is a sad hack 🙁

Reply:
IMO it's ok here, if not, please propose an alternative. For me enumerating them all would be even worse. The solution would be to refactor `distributed_helper`, but it is out of scope.